src/WebJobs.Extensions.DurableTask/AzureStorageDurabilityProviderFactory.cs

```diff
@@ -27,12 +28,21 @@ internal class AzureStorageDurabilityProviderFactory : IDurabilityProviderFactor
             IOptions<DurableTaskOptions> options,
             IConnectionStringResolver connectionStringResolver,
             INameResolver nameResolver,
-            ILoggerFactory loggerFactory)
+            ILoggerFactory loggerFactory,
+            IPlatformInformationService platformInfo = null)
         {
             this.options = options.Value;
             this.nameResolver = nameResolver;
             this.loggerFactory = loggerFactory;
             this.azureStorageOptions = new AzureStorageOptions();
+            this.inConsumption = platformInfo != null ? platformInfo.InConsumption() : false;
```

Comment:
Alternatively, we could just fail fast when a platformInfo object isn't provided. Looking back, I would prefer this alternative. What do others think?

Reply:
I second fail fast. It should never be null, since we provide the implementation.